libfwupdplugin/fu-device.c

```diff
@@ -4068,6 +4092,47 @@ fu_device_set_update_image(FuDevice *self, const gchar *update_image)
 	g_object_notify(G_OBJECT(self), "update-image");
 }
 
+/**
+ * fu_device_get_fwupd_version:
+ * @self: a #FuDevice
+ *
+ * Gets the fwupd version that created the emulation.
+ *
+ * Returns: a version, e.g. `2.0.8`, or %NULL if unset
+ *
+ * Since: 2.0.8
+ **/
+const gchar *
+fu_device_get_fwupd_version(FuDevice *self)
+{
+	FuDevicePrivate *priv = GET_PRIVATE(self);
+	g_return_val_if_fail(FU_IS_DEVICE(self), NULL);
+	return priv->fwupd_version;
+}
+
+/**
+ * fu_device_set_fwupd_version:
+ * @self: a #FuDevice
+ * @fwupd_version: (nullable): the version, e.g. `2.0.8`
+ *
+ * Sets the fwupd version that created the emulation.
+ *
+ * Since: 2.0.8
```

Comment:
This is oriented around emulation, but it actually applies to the full device.   If it's only intended to be used with emulated devices it probably needs a guard that the device has an emulated flag.

Reply:
That's a good idea, fixed.